integration-tests/client/test/core.test.ts

```diff
@@ -0,0 +1,355 @@
+    }, 120_000);
+
+    test("Liquid Stake Mint and Transfer", async () => {
+        const stridejs = strideAccounts.user;
+        const gaiajs = gaiaAccounts.user;
+        const stakeAmount = 10000000;
+
+        // Get initial balances
+        const { balance: { amount: strideInitialAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: ATOM_DENOM_ON_STRIDE,
+            });
+
+        const { balance: { amount: strideInitialStAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: "stuatom", // stATOM denom
+            });
+
+        // Get delegation address
+        const { hostZone } = await stridejs.query.stride.stakeibc.hostZone({
+            chainId: GAIA_CHAIN_ID,
+        });
+        const delegationAddress = hostZone?.delegationIcaAddress || "";
+
+        // Get initial delegation ICA balance
+        const delegationInitialBalance = await gaiajs.query.bank.balance(delegationAddress, UATOM);
+
+        console.log("Initial balances:");
+        console.log(`Stride ATOM: ${strideInitialAtomBalance}`);
+        console.log(`Stride stATOM: ${strideInitialStAtomBalance}`);
+        console.log(`Delegation ICA ATOM: ${delegationInitialBalance.amount}`);
+
+        // Perform liquid staking
+        const liquidStakeMsg = stride.stakeibc.MessageComposer.withTypeUrl.liquidStake({
+            creator: stridejs.address,
+            amount: String(stakeAmount),
+            hostDenom: UATOM,
+        });
+
+        await submitTxAndExpectSuccess(stridejs, [liquidStakeMsg]);
+
+        // Wait for stTokens to be minted
+        let strideFinalStAtomBalance;
+        let attempts = 0;
+        const maxAttempts = 60; // 30 seconds max wait time
+
+        while (attempts < maxAttempts) {
+            const { balance: { amount: currentStAtomBalance } = { amount: "0" } } =
+                await stridejs.query.cosmos.bank.v1beta1.balance({
+                    address: stridejs.address,
+                    denom: "stuatom",
+                });
+
+            if (BigInt(currentStAtomBalance) > BigInt(strideInitialStAtomBalance)) {
+                strideFinalStAtomBalance = currentStAtomBalance;
+                break;
+            }
+
+            attempts++;
+            await sleep(500);
+        }
+
+        if (attempts >= maxAttempts) {
+            throw new Error("Timed out waiting for stATOM tokens to be minted");
+        }
+
+        // Get final ATOM balance on Stride
+        const { balance: { amount: strideFinalAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: ATOM_DENOM_ON_STRIDE,
+            });
```

Comment:
similarly here, let's throw this into a helper so it  doesn't hurt the tests readability. Ideally we can just do something like:

```
const strideFinalStAtomBalance = await utils.waitForBalanceChange(...)
```

Reply:
if you want to re-use it for the while loop below too, maybe you can take it one step further and have like a `waitForStateChange` generic function that takes in a function as an argument that it retries. So in the above, you could pass the the get stride balance function and have it retry it until the response changes. And then below you can repeat the same thing but with a get delegation balance function

Possible this would be too much and muddy up the code, but worth a try

End goal is to keep the code in the actual test functions nice and concise so you can quickly see what's going on!